File: Chapter14/R/Example5.R
# Chapter 14 - Παραμετρικοί Έλεγχοι για Δύο Πληθυσμούς και ANOVA
# Example 5: One-Way ANOVA — Πλήρης Ανάλυση (διαφήμιση)

# Dataset
advertising <- data.frame(
  Sales    = c(23, 25, 27, 24, 26,
               30, 32, 28, 31, 29,
               20, 22, 19, 21, 20),
  Strategy = factor(rep(c("Television", "Internet", "Print"), each = 5))
)

# --- Descriptive statistics ---
library(dplyr)
desc <- advertising %>%
  group_by(Strategy) %>%
  summarise(
    n    = n(),
    Mean = mean(Sales),
    SD   = sd(Sales),
    Min  = min(Sales),
    Max  = max(Sales)
  )
print(desc)

# --- Visualization ---
library(ggplot2)
ggplot(advertising, aes(x = Strategy, y = Sales, fill = Strategy)) +
  geom_boxplot(alpha = 0.7, width = 0.5) +
  geom_jitter(width = 0.1, alpha = 0.5, size = 2) +
  stat_summary(fun = mean, geom = "point", shape = 23,
               size = 4, fill = "white") +
  labs(
    title    = "Sales by Advertising Strategy",
    subtitle = "Diamonds indicate group means",
    y = "Sales (thousands EUR)", x = "Strategy"
  ) +
  theme_minimal(base_size = 13) + theme(legend.position = "none")

# --- One-Way ANOVA ---
anova_model   <- aov(Sales ~ Strategy, data = advertising)
anova_summary <- summary(anova_model)
print(anova_summary)

# Extract key values
f_stat  <- anova_summary[[1]]$`F value`[1]
p_value <- anova_summary[[1]]$`Pr(>F)`[1]
SSB     <- anova_summary[[1]]$`Sum Sq`[1]
SST     <- sum(anova_summary[[1]]$`Sum Sq`)
cat("\nF-statistic:", round(f_stat, 3), "\n")
cat("p-value    :", format(p_value, scientific = TRUE), "\n")

# --- Effect size ---
eta_sq  <- SSB / SST
omega_sq <- (SSB - (2-1)*anova_summary[[1]]$`Mean Sq`[2]) /
            (SST + anova_summary[[1]]$`Mean Sq`[2])
cat("eta-squared  :", round(eta_sq, 4),
    "(", ifelse(eta_sq < 0.06, "small",
         ifelse(eta_sq < 0.14, "medium", "large")), ")\n")
cat("omega-squared:", round(omega_sq, 4), "\n")

# --- Decision ---
if (p_value < 0.05) {
  cat("\nDecision: Reject H0. At least one group mean differs.\n")
} else {
  cat("\nDecision: Fail to reject H0.\n")
}

File: Chapter14/Python/Example5.py
# Chapter 14 - Παραμετρικοί Έλεγχοι για Δύο Πληθυσμούς και ANOVA
# Example 5: One-Way ANOVA — Πλήρης Ανάλυση (διαφήμιση)

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats
from statsmodels.formula.api import ols
import statsmodels.api as sm

# Dataset
advertising = pd.DataFrame({
    'Sales'   : [23,25,27,24,26, 30,32,28,31,29, 20,22,19,21,20],
    'Strategy': ['Television']*5 + ['Internet']*5 + ['Print']*5
})

# --- Descriptive statistics ---
print(advertising.groupby('Strategy')['Sales']
      .agg(['count','mean','std','min','max'])
      .round(3))

# --- Visualization ---
fig, ax = plt.subplots(figsize=(9, 6))
sns.boxplot(x='Strategy', y='Sales', data=advertising,
            palette='Blues', width=0.4, ax=ax)
sns.stripplot(x='Strategy', y='Sales', data=advertising,
              color='steelblue', alpha=0.5, size=5, ax=ax)
means = advertising.groupby('Strategy')['Sales'].mean()
for i, (strat, m) in enumerate(means.items()):
    ax.scatter(i, m, marker='D', s=80, color='white',
               edgecolors='black', zorder=5)
ax.set_title('Sales by Advertising Strategy')
ax.set_ylabel('Sales (thousands EUR)')
plt.tight_layout()
plt.savefig('anova_boxplots.png', dpi=300, bbox_inches='tight')

# --- One-Way ANOVA via scipy ---
groups = [advertising[advertising['Strategy'] == s]['Sales'].values
          for s in ['Television', 'Internet', 'Print']]
f_stat, p_value = stats.f_oneway(*groups)
print(f"\nF-statistic = {f_stat:.3f}, p-value = {p_value:.6f}")

# --- ANOVA table via statsmodels ---
model       = ols('Sales ~ C(Strategy)', data=advertising).fit()
anova_table = sm.stats.anova_lm(model, typ=1)
print("\nANOVA Table:\n", anova_table.round(4))

# --- Effect size ---
SSB    = anova_table['sum_sq']['C(Strategy)']
SST    = anova_table['sum_sq'].sum()
MSW    = anova_table['mean_sq']['Residual']
eta_sq = SSB / SST
k, N   = 3, len(advertising)
omega_sq = (SSB - (k-1)*MSW) / (SST + MSW)
print(f"\neta-squared   = {eta_sq:.4f}")
size_label = "small" if eta_sq < 0.06 else "medium" if eta_sq < 0.14 else "large"
print(f"Interpretation: {size_label} effect")
print(f"omega-squared = {omega_sq:.4f}")

# --- Decision ---
print("\nDecision:", "Reject H0" if p_value < 0.05 else "Fail to reject H0")
